Library/Homebrew/keg.rb

```diff
@@ -295,14 +295,24 @@ def lock
   end
 
   def completion_installed?(shell)
-    dir = case shell
-    when :bash then path.join("etc", "bash_completion.d")
-    when :zsh  then path.join("share", "zsh", "site-functions")
-    when :fish then path.join("share", "fish", "vendor_completions.d")
+    case shell
+    when :bash
+      dir = path.join("etc", "bash_completion.d")
+    when :zsh
+      # By convention, zsh completion files start with an underscore
+      return !Dir["#{path}/share/zsh/site-functions/_*"].empty?
```

Comment:
The zsh case is looking for specific files in a directory, while the other shell cases are just looking for a non-empty directory.

I'm not sure how I might rework this so they all use the logic below.  Do you have a suggestion?